### PR TITLE
Better error display, adapt to latest server changes reporting subsystem state

### PR DESCRIPTION
**Pull request by rpiaggio**

### CODE REVIEW

File: modules/model/shared/src/main/scala/observe/model/ExecutionState.scala [hunk 1]
```diff
@@ -28,7 +26,7 @@ case class ExecutionState(
   observer:        Option[Observer],
   runningStepId:   Option[Step.Id],
   nsState:         Option[NsRunningState],
-  stepResources:   List[(Step.Id, List[(Resource | Instrument, ActionStatus)])],
+  stepResources:   Map[Step.Id, Map[Resource | Instrument, ActionStatus]],
```
Comment (rpiaggio) [inline modules/model/shared/src/main/scala/observe/model/ExecutionState.scala:29]: I found it easier to deal with this if we model as a `Map` (allows for example `index`/`at` optics).

File: modules/model/shared/src/main/scala/observe/model/ExecutionState.scala [hunk 2]
```diff
@@ -42,25 +40,12 @@ case class ExecutionState(
...
+  val nsState: Lens[ExecutionState, Option[NsRunningState]]                                       = Focus[ExecutionState](_.nsState)
+  val stepResources: Lens[ExecutionState, Map[Step.Id, Map[Resource | Instrument, ActionStatus]]] =
     Focus[ExecutionState](_.stepResources)
-  val systemOverrides: Lens[ExecutionState, SystemOverrides] =
+  val systemOverrides: Lens[ExecutionState, SystemOverrides]                                      =
     Focus[ExecutionState](_.systemOverrides)
-  val breakpoints: Lens[ExecutionState, Set[Step.Id]]        = Focus[ExecutionState](_.breakpoints)
-  val observer: Lens[ExecutionState, Option[Observer]]       = Focus[ExecutionState](_.observer)
-
-  def stepResourcesT(
```
Comment (rpiaggio) [inline modules/model/shared/src/main/scala/observe/model/ExecutionState.scala:56]: By using `index`/`at` this was left unused so I removed it.

File: modules/model/shared/src/main/scala/observe/model/StepState.scala [hunk 3]
```diff
@@ -16,35 +16,35 @@ enum StepState derives Eq:
   case Running             extends StepState
   case Paused              extends StepState
 
-  def canSetBreakpoint: Boolean = this match
+  lazy val canSetBreakpoint: Boolean = this match
```
Comment (rpiaggio) [inline modules/model/shared/src/main/scala/observe/model/StepState.scala:19]: I was having a strange lazy initialization error regarding the derived `Eq`. Thought changing to `lazy val`s would fix it, but it didn't (`sbt clean` did). I decided to leave these as `lazy val`s anwyay.

File: modules/web/client/src/main/webapp/styles/observe.scss [hunk 4]
```diff
@@ -166,58 +166,71 @@ html {
   }
 }
 
-// .ObserveStyles-rowPositive {
-//   background-color: @positiveBackgroundColor;
-//   color: @positiveTextColor;
-//   box-shadow: @positiveBoxShadow;
-// }
+.pl-react-table.p-datatable.p-datatable-hoverable-rows .p-datatable-tbody {
+  & > tr {
```
Comment (rpiaggio) [inline modules/web/client/src/main/webapp/styles/observe.scss:170]: Commented styles are still pending revision from seqexec migration.

File: modules/model/shared/src/main/scala/observe/model/ExecutionState.scala (hunk 1 above)
Comment (cquiroz) [reply modules/model/shared/src/main/scala/observe/model/ExecutionState.scala:29]: 👍

File: modules/model/shared/src/main/scala/observe/model/StepState.scala (hunk 3 above)
Comment (cquiroz) [reply modules/model/shared/src/main/scala/observe/model/StepState.scala:19]: I wonder if these should be `inline`

**Review by cquiroz — APPROVED** [review]
LGTM

**Pull request merged**